test/math_libraries/convolutions.py

```diff
@@ -0,0 +1,113 @@
+import torch
+from torch.testing._internal.common_device_type import instantiate_device_type_tests, \
+    dtypes, dtypesIfCUDA, skipCUDAIfRocm, onlyOnCPUAndCUDA
+from torch.testing._internal.common_nn import NNTestCase
+from torch.testing._internal.common_utils import run_tests
+import os
+import json
+
+class TestConvExt(NNTestCase):
+    def _subtensor(self, tensor, dim, group, g):
+        if tensor is None:
+            return None
+        group_size = int(tensor.size(dim) / group)
+        return tensor.narrow(dim, group_size * g, group_size).contiguous()
+
+    def _thnn_conv(self, input, weight, k, bias, s, p, d):
+        if d[0] > 1 or d[1] > 1:
+            return torch._C._nn.slow_conv_dilated2d(input, weight, k, bias, s, p, d)
+        else:
+            return torch._C._nn.thnn_conv2d(input, weight, k, bias, s, p)
+
+    def _thnn_conv_group(self, input, weight, k, bias, s, p, d, group):
+        if group == 1:
+            return self._thnn_conv(input, weight, k, bias, s, p, d)
+        else:
+            outputs = []
+            for g in range(group):
+                input_g = self._subtensor(input, 1, group, g)
+                weight_g = self._subtensor(weight, 0, group, g)
+                bias_g = self._subtensor(bias, 0, group, g)
+                outputs.append(self._thnn_conv(input_g, weight_g, k, bias_g, s, p, d))
+            return torch.cat(outputs, 1)
+
+    def _collect_cases(self):
```

Comment:
This can acquire unexpected files, a security concern. It needs to explicitly acquire the desired files or take in an argument, like an environment variable, specifying the files. Otherwise someone can just put a certain type of file in the directory and this test will try and load it.